src/components/AssetBreadcrumb/AssetBreadcrumb.tsx

```diff
@@ -0,0 +1,166 @@
+import { Asset } from '@cognite/sdk';
+import { Breadcrumb, Icon } from 'antd';
+import React, { useContext, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { ERROR_NO_SDK_CLIENT } from '../../constants/errorMessages';
+import { ClientSDKCacheContext } from '../../context/clientSDKCacheContext';
+import { withDefaultTheme } from '../../hoc/withDefaultTheme';
+import { AnyIfEmpty } from '../../interfaces';
+import { defaultTheme } from '../../theme/defaultTheme';
+
+export interface AssetBreadcrumbProps {
+  assetId: number;
+  maxLength?: number;
+  renderItem?: (asset: Asset, depth: number) => JSX.Element;
+  onBreadcrumbClick?: (asset: Asset, depth: number) => void;
+  theme?: AnyIfEmpty<{}>;
+}
+
+const ENTER_KEY_CODE = 13;
+
+const renderBreadcrumbs = (
+  assets: Asset[],
+  renderAsset: (asset: Asset, depth: number) => JSX.Element,
+  maxLength: number = 3
```

Comment:
I think it is strange to have a default limit here. Especially when it is so small. Maybe 10 / 25 is better?

Reply:
we can set default value greater, but guess not more than 7-8. Overall, you need to have a parameter to control length of assets chain depending on place where you add gearbox component (header, sidebar, etc.)